_sub/database/postgres/main.tf

```diff
@@ -36,6 +36,10 @@ resource "aws_db_parameter_group" "dbparams" {
     value = "1"
   }
 
+  lifecycle {
+    create_before_destroy = true
```

Comment:
One concern: "Destroy provisioners of this resource will not run if create_before_destroy is set to true" 

Ideally we would want to make this lifecycle change to be optional, and it should rather be controlled through parameters, like:

`create_before_destroy = var.create_before_destroy`

In _sub/database/postgres/vars.tf then you also define this create_before_destroy bool variable with a default value of true

Reply:
The security group defined above the parameter group has the same lifecycle defined. Should that also reference the 
`var.create_before_destroy` parameter I will create.

Reply:
Since you did not add that to the security group, I would say it would be scope creep for now. The change looks good to me as it is now.